chain/xrp/client/client.go

```diff
@@ -0,0 +1,275 @@
+	Index             string `json:"Index"`
+}
+
+// FetchTransferInput returns tx input for a Template tx
+func (client *Client) FetchTransferInput(ctx context.Context, args xcbuilder.TransferArgs) (xc.TxInput, error) {
+	return &tx_input.TxInput{}, errors.New("not implemented")
+}
+
+// Deprecated method - use FetchTransferInput
+func (client *Client) FetchLegacyTxInput(ctx context.Context, from xc.Address, to xc.Address) (xc.TxInput, error) {
+	// No way to pass the amount in the input using legacy interface, so we estimate using min amount.
+	args, _ := xcbuilder.NewTransferArgs(from, to, xc.NewAmountBlockchainFromUint64(1))
+	return client.FetchTransferInput(ctx, args)
+}
+
+// SubmitTx submits a Template tx
+func (client *Client) SubmitTx(ctx context.Context, txInput xc.Tx) error {
+	return errors.New("not implemented")
+}
+
+// Returns transaction info - legacy/old endpoint
+func (client *Client) FetchLegacyTxInfo(ctx context.Context, txHash xc.TxHash) (xc.LegacyTxInfo, error) {
+	return xc.LegacyTxInfo{}, errors.New("not implemented")
+}
+
+// Returns transaction info - new endpoint
+func (client *Client) FetchTxInfo(ctx context.Context, txHash xc.TxHash) (xclient.TxInfo, error) {
+	return xclient.TxInfo{}, errors.New("not implemented")
+}
+
+// FetchBalance fetches token balance for a XRP address
+func (client *Client) FetchBalance(ctx context.Context, address xc.Address) (xc.AmountBlockchain, error) {
+	return client.FetchBalanceForAsset(ctx, address, client.Asset)
+}
+
+func (client *Client) FetchBalanceForAsset(ctx context.Context, address xc.Address, assetCfg xc.ITask) (xc.AmountBlockchain, error) {
+	switch asset := assetCfg.(type) {
+	case *xc.ChainConfig:
+		return client.FetchNativeBalance(ctx, address)
+	case *xc.TokenAssetConfig:
+		return client.fetchContractBalance(ctx, address, asset.Contract)
+	default:
+		contract := asset.GetContract()
+		logrus.WithFields(logrus.Fields{
+			"chain":      asset.GetChain().Chain,
+			"contract":   contract,
+			"asset_type": fmt.Sprintf("%T", asset),
+		}).Warn("fetching balance for unknown asset type")
+		return client.fetchContractBalance(ctx, address, contract)
+	}
+}
+
+// FetchNativeBalance fetches account native balance for a XRP address
+func (client *Client) FetchNativeBalance(ctx context.Context, address xc.Address) (xc.AmountBlockchain, error) {
+	zero := xc.NewAmountBlockchainFromUint64(0)
+
+	requestPayload := AccountInfoRequest{
+		Method: "account_info",
+		Params: []ParamEntry{
+			{
+				Account: address,
+			},
+		},
+	}
+
+	jsonPayload, err := json.Marshal(requestPayload)
+	if err != nil {
+		return zero, fmt.Errorf("failed to marshal request payload: %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "POST", client.Url, bytes.NewBuffer(jsonPayload))
+	if err != nil {
+		return zero, fmt.Errorf("failed to create new HTTP request: %w", err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	httpClient := http.DefaultClient
```

Comment:
This should be done in the `Client` constructor.   Then would use `client.httpClient`.